chain/deals/provider.go

```diff
@@ -192,6 +206,44 @@ func (p *Provider) onUpdated(ctx context.Context, update minerDealUpdate) {
 	}
 }
 
+// onDataTransferEvent is the function called when an event occurs in a data
+// transfer -- it reads the voucher to verify this even occurred in a storage
+// market deal, then, based on the data transfer event that occurred, it generates
+// and update message for the deal -- either moving to staged for a completion
+// event or moving to error if a data transfer error occurs
+func (p *Provider) onDataTransferEvent(event datatransfer.Event, channelState datatransfer.ChannelState) {
+	voucher, ok := channelState.Voucher().(*StorageDataTransferVoucher)
+	// if this event is for a transfer not related to storage, ignore
+	if !ok {
+		return
```

Comment:
at least log a warning "got wrong type of voucher"

Reply:
it's not really a warning per se -- data transfer event is all events on data transfer. there could be retrieval events coming through too -- so it really just means this isn't for me.